src/nsls2api/api/v1/proposal_api.py

```diff
@@ -118,6 +118,29 @@ async def get_proposals(
     return response_model
 
 
+@router.get("/proposal/saf/{saf_id}", response_model=SingleProposal)
+async def get_proposal_by_saf(saf_id: str):
+    try:
+        proposal = await proposal_service.proposal_by_saf_id(saf_id)
+        if proposal is None:
+            raise HTTPException(
+                status_code=fastapi.status.HTTP_404_NOT_FOUND,
+                detail=f"Proposal with SAF {saf_id} not found",
+            )
```

Comment:
This branch will never get called.  You could cut this in favor of the `LookupError` exception below.